ports/nrf/modules/machine/uart.c

```diff
@@ -293,6 +335,58 @@ static bool mp_machine_uart_txdone(machine_uart_obj_t *self) {
     return !nrfx_uart_tx_in_progress(self->p_uart);
 }
 
+#if MICROPY_PY_MACHINE_UART_IRQ
+static mp_uint_t uart_irq_trigger(mp_obj_t self_in, mp_uint_t new_trigger) {
+    machine_uart_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    self->mp_irq_trigger = new_trigger;
+    return 0;
+}
+
+static mp_uint_t uart_irq_info(mp_obj_t self_in, mp_uint_t info_type) {
+    machine_uart_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    if (info_type == MP_IRQ_INFO_FLAGS) {
+        return self->mp_irq_flags;
+    } else if (info_type == MP_IRQ_INFO_TRIGGERS) {
+        return self->mp_irq_trigger;
+    }
+    return 0;
+}
+
+const mp_irq_methods_t uart_irq_methods = {
+    .trigger = uart_irq_trigger,
+    .info = uart_irq_info,
+};
+
+static mp_irq_obj_t *mp_machine_uart_irq(machine_uart_obj_t *self, bool any_args, mp_arg_val_t *args) {
+    if (self->mp_irq_obj == NULL) {
+        self->mp_irq_trigger = 0;
+        self->mp_irq_obj = mp_irq_new(&uart_irq_methods, MP_OBJ_FROM_PTR(self));
+        MP_STATE_PORT(nrf_uart_irq_obj) = self->mp_irq_obj;
```

Comment:
In case one day more UARTs are supported by this port, this should be generalised to an array, and do something like:
```c
MP_STATE_PORT(nrf_uart_irq_obj)[self->uart_id] = self->mp_irq_obj;
```

Reply:
There is no uart_id element in the `machine_uart_obj_t` data type. I can add it, or use a constant index of 0.

Reply:
Added uart_id.